tutorial/markdown/nodejs/quickstart-typescript/quickstart-typescript.md

```diff
@@ -0,0 +1,483 @@
+
+
+## Prerequisites
+
+To run this prebuilt project, you will need:
+
+- [Couchbase Capella](https://www.couchbase.com/products/capella/) cluster with [travel-sample](https://docs.couchbase.com/nodejs-sdk/current/ref/travel-app-data-model.html) bucket loaded.
+  - To run this tutorial using a self managed Couchbase cluster, please refer to the [appendix](#running-self-managed-couchbase-cluster).
+- [LTS Node.js Version](https://nodejs.org/en/download)
+- Loading Travel Sample Bucket
+  If travel-sample is not loaded in your Capella cluster, you can load it by following the instructions for your Capella Cluster:
+
+  - [Load travel-sample bucket in Couchbase Capella](https://docs.couchbase.com/cloud/clusters/data-service/import-data-documents.html#import-sample-data)
+
+> Note that this tutorial is designed to work with the latest Node SDK version (4.x) for Couchbase. It will not work with the older Node js versions  for Couchbase without adapting the code.
+
+### Couchbase Capella Configuration
+
+When running Couchbase using [Capella](https://cloud.couchbase.com/), the following prerequisites need to be met.
+
+- The application requires the travel-sample bucket to be [loaded](https://docs.couchbase.com/cloud/clusters/data-service/import-data-documents.html#import-sample-data) in the cluster from the Capella UI.
+- Create the [database credentials](https://docs.couchbase.com/cloud/clusters/manage-database-users.html) to access the travel-sample bucket (Read and Write) used in the application.
+- [Allow access](https://docs.couchbase.com/cloud/clusters/allow-ip-address.html) to the Cluster from the IP on which the application is running.
+## App Setup
+### Cloning Repo
+
+```shell
+git clone https://github.com/couchbase-examples/typescript-quickstart.git
+```
+
+### Install Dependencies
+
+Any dependencies will be installed by running the npm install command from the root directory of the project.
+
+```shell
+npm install
+```
+### Setup Database Configuration
+
+To know more about connecting to your Capella cluster, please follow the [instructions](https://docs.couchbase.com/cloud/get-started/connect.html).
+
+Specifically, you need to do the following:
+
+- Create the [database credentials](https://docs.couchbase.com/cloud/clusters/manage-database-users.html) to access the travel-sample bucket (Read and Write) used in the application.
+- [Allow access](https://docs.couchbase.com/cloud/clusters/allow-ip-address.html) to the Cluster from the IP on which the application is running.
+
+All configuration for communication with the database is read from the environment variables. We have provided a convenience feature in this quickstart to read the environment variables from a local file, `dev.env` in the `config` folder.
+
+Create a copy of `config/dev.env.example` file and rename it to `config/dev.env` and add the values for the Couchbase connection.
+
+
+```sh
```

Comment:
This is inconsistent with the repo.

Reply:
I will update the repo readme.

Reply:
This is still inconsistent
https://github.com/couchbase-examples/typescript-quickstart/blob/main/config/dev.env.example

Reply:
Yep, changed it.